pubtools/pulplib/_impl/model/maintenance.py

```diff
@@ -0,0 +1,200 @@
+import logging
+import datetime
+import os
+
+import jsonschema
+
+from pubtools.pulplib._impl import compat_attr as attr
+from ..schema import load_schema
+from .common import InvalidDataException
+
+
+LOG = logging.getLogger("pubtools.pulplib")
+
+
+USER = os.environ.get("USER")
+HOSTNAME = os.environ.get("HOSTNAME")
+
+
+def iso_time_now():
+    return datetime.datetime.utcnow().strftime("%Y-%m-%dT%H:%M:%SZ")
+
+
+@attr.s(kw_only=True, frozen=True)
+class MaintenanceEntry(object):
+    """Details about the maintenance status of a specific repository."""
+
+    repo_id = attr.ib(type=str)
+    """ID of repository in maintenance.
+    Note: there is no guarantee that a repository of this ID currenly exists
+          in the Pulp server."""
+    message = attr.ib(default=None, type=str)
+    """Why this repository is in maintenance."""
+    owner = attr.ib(default=None, type=str)
+    """Who set this repository in maintenance mode."""
+    started = attr.ib(default=iso_time_now(), type=datetime.datetime)
```

Comment:
I was reviewing docs and noticed something: due to this default, the documentation for this class actually contains the time at which the docs were generated :)

At first I thought it's just a funny thing, but thinking about it some more - this shows that the default is a bad idea, I think.  The reason is that a piece of code creating a MaintenanceEntry will create different objects depending on when that code is run.

For example:

```
   entry1 = MaintenanceEntry(repo_id='repo1', owner='some-user')
   entry2 = MaintenanceEntry(repo_id='repo1', owner='some-user')

   # question: is this true or false?
   # answer: depends how much time passed between above two lines!
   assert entry1 == entry2
```

I would say it violates reasonable expectations for this kind of "plain old data" class - two identical pieces of code for creating an object, should give you two identical objects, and not have a hidden dependency on the clock.

So, could we please set the default here to None, and handle this some other way?  e.g. setting the started value during `add`, if needed.

(If there's a question "why would anyone manually create a MaintenanceEntry anyway" - I'd expect the main reason to do that is when writing autotests related to maintenance mode features.)